dandi/publish/tests/test_asset.py

```diff
@@ -73,3 +73,44 @@ def test_asset_rest_retrieve(api_client, asset):
         'updated': TIMESTAMP_RE,
         'metadata': asset.metadata,
     }
+
+
+@pytest.mark.django_db
+def test_asset_path_filter(api_client, asset):
+    asset_data = {
+        'version': {
+            'dandiset': {
+                'identifier': asset.version.dandiset.identifier,
+                'created': TIMESTAMP_RE,
+                'updated': TIMESTAMP_RE,
+            },
+            'version': asset.version.version,
+            'name': asset.version.name,
+            'description': asset.version.description,
+            'created': TIMESTAMP_RE,
+            'updated': TIMESTAMP_RE,
+            'count': 1,
+        },
+        'uuid': str(asset.uuid),
+        'path': asset.path,
+        'size': asset.size,
+        'sha256': asset.sha256,
+        'created': TIMESTAMP_RE,
+        'updated': TIMESTAMP_RE,
+        'metadata': asset.metadata,
+    }
+
+    partial_path_assets = api_client.get(
+        f'/api/dandisets/{asset.version.dandiset.identifier}/'
+        f'versions/{asset.version.version}/assets/',
+        {'path': asset.path[: int(len(asset.path) / 2)]},
+    ).data['results']
+
+    full_path_assets = api_client.get(
+        f'/api/dandisets/{asset.version.dandiset.identifier}/'
+        f'versions/{asset.version.version}/assets/',
+        {'path': asset.path},
+    ).data['results']
+
+    assert asset_data in partial_path_assets
```

Comment:
Is it possible to reduce what is being asserted, allowing us to remove the giant `asset_data` variable? I think just asserting that the `asset.id` was in the returned data is adequate.